### ISSUE TITLE
Bug: Error: Could not find config file.

### ISSUE THREAD

**Issue by phawxby** [eslint/eslint#17784]
### Environment

Node version: v16.20.0
npm version: v9.6.6
Local ESLint version: Not found
Global ESLint version: Not found
Operating System: darwin 23.0.0

### What parser are you using?

Other

### What did you do?

Large turborepo monorepo project with no root configuration file but a config file per package. 

```
└── project/
    ├── apps/
    │   └── app-1/
    │       ├── ...
    │       └── eslint.config.js
    ├── packages/
    │   └── package-1/
    │       └── eslint.config.js
```

`vscode-eslint` plugin cannot find a config file. More detailed breakdown of the issue is here.
https://github.com/microsoft/vscode-eslint/issues/1737#issuecomment-1820582808

### What did you expect to happen?

Per-package config file resolution.

### What actually happened?

The path of the file being linted is ignored and config file is searched for in `cwd`, the root of the project. 

Below is the `vscode-eslint` output however the underlying issue appears to [reside here](https://github.com/eslint/eslint/blob/de165c108203c6703516ac651f5b4cac5b241804/lib/eslint/flat-eslint.js#L941), the current file path is not included as part of config file resolution.

```
[Error - 10:57:17] An unexpected error occurred:
[Error - 10:57:17] Error: Could not find config file.
    at locateConfigFileToUse (/Users/paulhawxby/Documents/GitHub/eslint-flat-bug/node_modules/.pnpm/eslint@8.54.0/node_modules/eslint/lib/eslint/flat-eslint.js:332:21)
    at async calculateConfigArray (/Users/paulhawxby/Documents/GitHub/eslint-flat-bug/node_modules/.pnpm/eslint@8.54.0/node_modules/eslint/lib/eslint/flat-eslint.js:367:49)
    at async FlatESLint.lintText (/Users/paulhawxby/Documents/GitHub/eslint-flat-bug/node_modules/.pnpm/eslint@8.54.0/node_modules/eslint/lib/eslint/flat-eslint.js:941:25)
    at async /Users/paulhawxby/.vscode/extensions/dbaeumer.vscode-eslint-2.4.2/server/out/eslintServer.js:1477:27
    at async E (/Users/paulhawxby/.vscode/extensions/dbaeumer.vscode-eslint-2.4.2/server/out/eslintServer.js:1141:24)
    at async /Users/paulhawxby/.vscode/extensions/dbaeumer.vscode-eslint-2.4.2/server/out/eslintServer.js:12289:21
```

### Link to Minimal Reproducible Example

https://github.com/phawxby/eslint-flat-bug

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

This appears to have been reported previously but not resolved #16485

**Comment by Rec0iL99** [comment]
Hi @phawxby, thanks for the issue. I believe this is working as intended. In the flat config system, `eslint.config.js` is [resolved](https://eslint.org/docs/latest/use/configure/configuration-files-new#configuration-file) relative to the `cwd`. For module-specific configs, please use glob patterns to apply different configurations for different modules, instead of having a `eslint.config.js` per module.

```js
// project root eslint.config.js

module.exports = [
  {
    files: ["packages/package-1/**/*.js"],
    rules: {
      semi: "error",
    },
  },
  {
    files: ["packages/package-2/**/*.js"],
    rules: {
      "no-unused-vars": "error",
    },
  },
];
```

As for the error with '`vscode-eslint`, we are unable to assist you as it is an extension not maintained by the ESLint team. Please consult their support for help.
